File: server/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/tpa/TpaPlayerLookup.java
package calebxzhou.rdi.mc.common2.tpa;

import java.util.UUID;

public interface TpaPlayerLookup {
    TpaPlayer findByName(String name);

    TpaPlayer findById(UUID id);

    void teleportTo(TpaPlayer requester, TpaPlayer target);
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/tpa/TpaPlayerLookup.kt
package calebxzhou.rdi.mc.rcmd.tpa

import java.util.*

interface TpaPlayerLookup {
    fun findByName(name: String): TpaPlayer?

    fun findById(id: UUID): TpaPlayer?

    fun teleportTo(requester: TpaPlayer, target: TpaPlayer)
}
